### PR TITLE
Hero banner adjustments with Nisra example

### PR DESCRIPTION
**Pull request by trevorsaint**
Updates to the hero component include visual tweaks and alignment.

Nisra version added also.

### CODE REVIEW

**Review by jrbarnes9 — COMMENTED** [review]
- Remove the `u-mb-m` from the button in the 'default' example
- Should we ensure last element always has it's bottom margin removed? (see engagement examples)
- Are we able to prevent `button` being used with `censusThemeDark`? Because it fails contrast accessibility. Otherwise, maybe we should switch button style to `ghosted`.
Screenshot 2021-02-19 at 08 53 03

- Need to consider light theme being used without the warning panel following.  Being used a lot in campaigns, but doesn't look right (especially on mobile). Maybe we need to add a border? 
Screenshot 2021-02-19 at 08 50 59

- Image not found in full screen Nisra example...
Screenshot 2021-02-19 at 08 46 29

**Comment by jrbarnes9** [comment]
The preTitle logo looks bigger than the census logo and the H1 alignment is off

Screenshot 2021-02-19 at 10 21 48

**Comment by trevorsaint** [comment]
> The preTitle logo looks bigger than the census logo and the H1 alignment is off
> 
> Screenshot 2021-02-19 at 10 21 48

**Yes it does, and should not be, I will fix**

In regards heading alignment. With the way fonts are designed, more noticeable in headings you get this additional padding naturally. It also occurs with line height. Some solutions do exist for line height crop (https://medium.com/eightshapes-llc/cropping-away-negative-impacts-of-line-height-84d744e016ce), however the only solution I can see in this instance is negative left positioning for alignment. Unless you have a better solution?

**Review by jrbarnes9 — APPROVED** [review]

File: src/components/hero/_macro.njk [hunk 1]
```diff
@@ -53,24 +82,33 @@
...
             </svg>
```
Comment (rmccar) [inline src/components/hero/_macro.njk:101]: Icons macro should be used here

File: src/components/hero/_macro.njk [hunk 2]
```diff
@@ -53,24 +82,33 @@
...
             {% if params.censusThemeDark is not defined or params.censusThemeDark === false %}
-                <div class="hero__circle-white"></div>
+                <div class="hero__circle"></div>
             {% endif %}
+
         {% endif %}
+
     </div>
+
 </section>
```
Comment (rmccar) [inline src/components/hero/_macro.njk:113]: Indentation needs fixing here

File: src/components/hero/_macro.njk [hunk 3]
```diff
@@ -53,24 +82,33 @@
...
             </svg>
+
```
Comment (rmccar) [inline src/components/hero/_macro.njk:102]:
```suggestion

```

File: src/components/hero/_macro.njk [hunk 4]
```diff
@@ -53,24 +82,33 @@
...
             {% endif %}
+
             {% if params.suffixText is defined and params.suffixText %}
-                <p class="hero__suffixText u-fs-s u-mt-l u-mb-no">{{ params.suffixText | safe }}</p>
+                <p class="hero__suffixText u-fs-s u-mt-m u-mb-no">{{ params.suffixText | safe }}</p>
             {% endif %}
+
         </div>
         {% if params.censusTheme is defined and params.censusTheme %}
+
```
Comment (rmccar) [inline src/components/hero/_macro.njk:92]:
```suggestion

```

File: src/components/hero/_macro.njk [hunk 5]
```diff
@@ -1,14 +1,40 @@
...
+{# if hero style is a dark theme, change theme #}
+{% if params.censusThemeDark is defined and params.censusThemeDark %}
+    {% set detailsClasses = " hero__details--ondark" %}
+{% endif %}
+
+
 <section class="hero grid--gutterless{% if params.censusTheme is defined and params.censusTheme %} hero--census{% endif %}">
-    <div class="hero__container container">
-        <div class="hero__details grid__col{% if params.censusTheme is defined and params.censusTheme %} col-6@m{%else%} col-8@m{% endif%}{% if params.censusThemeDark is defined and params.censusThemeDark %} hero__details--ondark{% endif%}">
+
```
Comment (rmccar) [inline src/components/hero/_macro.njk:33]:
```suggestion

```

File: src/components/hero/_macro.njk [hunk 6]
```diff
@@ -1,14 +1,40 @@
...
+    {% set detailsClasses = " hero__details--ondark" %}
+{% endif %}
+
+
 <section class="hero grid--gutterless{% if params.censusTheme is defined and params.censusTheme %} hero--census{% endif %}">
-    <div class="hero__container container">
-        <div class="hero__details grid__col{% if params.censusTheme is defined and params.censusTheme %} col-6@m{%else%} col-8@m{% endif%}{% if params.censusThemeDark is defined and params.censusThemeDark %} hero__details--ondark{% endif%}">
+   
+    <div class="hero__container container{{ containerClasses }}">
+
```
Comment (rmccar) [inline src/components/hero/_macro.njk:35]:
```suggestion

```

File: src/components/hero/_macro.njk [hunk 7]
```diff
@@ -1,14 +1,40 @@
...
+
 <section class="hero grid--gutterless{% if params.censusTheme is defined and params.censusTheme %} hero--census{% endif %}">
-    <div class="hero__container container">
-        <div class="hero__details grid__col{% if params.censusTheme is defined and params.censusTheme %} col-6@m{%else%} col-8@m{% endif%}{% if params.censusThemeDark is defined and params.censusThemeDark %} hero__details--ondark{% endif%}">
+   
+    <div class="hero__container container{{ containerClasses }}">
+       
+        <div class="hero__details{{ detailsClasses }} grid__col col-{{ detailsColumns }}@m">
+
```
Comment (rmccar) [inline src/components/hero/_macro.njk:37]:
```suggestion

```

File: src/components/hero/_macro.njk [hunk 8]
```diff
@@ -1,14 +1,40 @@
 {% macro onsHero(params) %}
 
+
```
Comment (rmccar) [inline src/components/hero/_macro.njk:3]:
```suggestion
```

File: src/components/hero/_macro.njk [hunk 9]
```diff
@@ -1,14 +1,40 @@
...
+{# style adjustments if dark or light skin #}
 {% if params.censusThemeDark is defined and params.censusThemeDark %}
     {% set skinName = "dark" %}
 {% else %}
     {% set skinName = "light" %}
 {% endif %}
 
+
```
Comment (rmccar) [inline src/components/hero/_macro.njk:11]:
```suggestion
```

File: src/components/hero/_macro.njk [hunk 10]
```diff
@@ -1,14 +1,40 @@
...
+{# If hero contains a collapsible, style it differently #}
+{% if params.collapsible is defined and params.collapsible %}
+    {% set containerClasses = " hero__container--has-collapsible" %}
+{% endif %}
+
+
```
Comment (rmccar) [inline src/components/hero/_macro.njk:17]:
```suggestion
```

File: src/components/hero/_macro.njk [hunk 11]
```diff
@@ -1,14 +1,40 @@
...
+{# if hero is census specific, change column usage #}
+{% if params.censusTheme is defined and params.censusTheme %}
+    {% set detailsColumns = "6" %}
+{% else %}
+    {% set detailsColumns = "8" %}
+{% endif %}
+
+
```
Comment (rmccar) [inline src/components/hero/_macro.njk:25]:
```suggestion
```

File: src/components/hero/_macro.njk [hunk 12]
```diff
@@ -1,14 +1,40 @@
...
+{# if hero style is a dark theme, change theme #}
+{% if params.censusThemeDark is defined and params.censusThemeDark %}
+    {% set detailsClasses = " hero__details--ondark" %}
+{% endif %}
+
+
```
Comment (rmccar) [inline src/components/hero/_macro.njk:31]:
```suggestion
```

**Review by rmccar — COMMENTED** [review]
Bit nit picky but theres a load of whitespace that could do with being removed

File: src/components/hero/_macro.njk [hunk 13]
```diff
@@ -99,7 +96,13 @@
             <svg class="hero__circle-lined" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 243 243" focusable="false" aria-hidden="true">
```
Comment (rmccar) [inline src/components/hero/_macro.njk:96]: SVG still needs removing here and isnt currently showing image in example

image

Comment (trevorsaint) [reply src/components/hero/_macro.njk:96]: Let me take a look at this again.

Comment (trevorsaint) [reply src/components/hero/_macro.njk:96]: In regards the image, I think this is a CDN issue. The same approach has been used for the other examples.

**Review by rmccar — APPROVED** [review]

**Comment by jrbarnes9** [comment]
Percy?

**Review by rmccar — DISMISSED** [review]

**Review by jrbarnes9 — COMMENTED** [review]
- Engage examples are inonsistently using `subtitle` or `text` for the same thing
- There's a potential need to add links to the hero `text` on dark theme (for the next phase) - so can we include supporting a link in this PR? 
- We've convinced campaigns not to use light hero without warning panel now, so we can revert this to white bg

**Comment by trevorsaint** [comment]
> * Engage examples are inonsistently using `subtitle` or `text` for the same thing
> * There's a potential need to add links to the hero `text` on dark theme (for the next phase) - so can we include supporting a link in this PR?
> * We've convinced campaigns not to use light hero without warning panel now, so we can revert this to white bg

Ok @jrbarnes9 I will look at these after the Welsh Audio work I have.

**Review by jrbarnes9 — APPROVED** [review]

**Review by rmccar — DISMISSED** [review]

**Review by boxadesign — APPROVED** [review]

**Review by jrbarnes9 — APPROVED** [review]

**Pull request merged**